mpmath/__init__.py

```diff
@@ -432,6 +436,34 @@
 unit_triangle = mp.unit_triangle
 sigmoid = mp.sigmoid
 
+
+class SetMPMathPrecisionError(RuntimeError):
+
+    def __init__(self, attribute):
+        self._attribute = attribute
+
+    def __str__(self):
+        name = self._attribute
+        return ("cannot set '%s' on 'mpmath'. Did you mean to "
+                "set '%s' on 'mpmath.mp'?" % (name, name))
+
+
+class _MPMathModuleType(types.ModuleType):
+
+    def _set_dps(self, value):
+        raise SetMPMathPrecisionError('dps')
+
+    dps = property(fset=_set_dps)
+
+    def _set_prec(self, value):
+        raise SetMPMathPrecisionError('prec')
+
+    prec = property(fset=_set_prec)
```

Comment:
I've not seen this in issues, but there are examples of setting also `mp.pretty attribute in docs.  Also, `mp.trap_complex` is mentioned.  I think we should add at least these properties as well.